### PR TITLE
feat: add cbh instance resource and instances data source

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add cbh instance resource and instances data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add cbh instance resource and instances data source
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=
'./huaweicloud/services/acceptance/cbh/' TESTARGS='-run TestAccCBHInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbh/ -v -run TestAccCBHInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccCBHInstance_basic
=== PAUSE TestAccCBHInstance_basic
=== CONT  TestAccCBHInstance_basic
--- PASS: TestAccCBHInstance_basic (1087.62s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbh       1087.678s

make testacc TEST=
'./huaweicloud/services/acceptance/cbh/' TESTARGS='-run TestAccDatasourceCbhInstances_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbh/ -v -run TestAccDatasourceCbhInstances_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceCbhInstances_basic 
=== PAUSE TestAccDatasourceCbhInstances_basic 
=== CONT  TestAccDatasourceCbhInstances_basic 
--- PASS: TestAccDatasourceCbhInstances_basic (1053.07s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbh       1053.145s
```

### CODE REVIEW

File: docs/resources/cbh_instance.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+  Changing this parameter will create a new resource.
+
+* `period_unit` - (Optional, String, ForceNew) Specifies the charging period unit of the instance. Valid values are
+  **month** and **year**. This parameter is mandatory if `charging_mode` is set to **prePaid**.
+
+  Changing this parameter will create a new resource.
+
+* `period` - (Optional, String, ForceNew) Specifies the charging period of the read replica instance. If `period_unit`
+  is set to **month**, the value ranges from 1 to 9. If `period_unit` is set to **year**, the value ranges from 1 to 3.
+  This parameter is mandatory if `charging_mode` is set to **prePaid**.
```
Comment (Jason-Zhang9309) [inline docs/resources/cbh_instance.md]: put this together with charging mode

File: docs/resources/cbh_instance.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+* `relative_resource_id` - (Optional, String, ForceNew) Specifies the new capacity expansion.
+
+  Changing this parameter will create a new resource.
+
+* `product_info` - (Optional, List, ForceNew)
+
+  Changing this parameter will create a new resource.
+  The [ProductInfo](#CBHInstance_ProductInfo) structure is documented below.
+
+* `order_id` - (Optional, String, ForceNew) Specifies the ID of the CBH order.
```
Comment (Jason-Zhang9309) [inline docs/resources/cbh_instance.md]: this parameter not used in the code

File: docs/resources/cbh_instance.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+
+  Changing this parameter will create a new resource.
+
+* `use_coupon` - (Optional, String, ForceNew) Specifies whether coupons are used for order payment.
```
Comment (Jason-Zhang9309) [inline docs/resources/cbh_instance.md]: this too

File: docs/resources/cbh_instance.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+
+  Changing this parameter will create a new resource.
+
+* `use_discount` - (Optional, String, ForceNew) Specifies whether a discount is used for the order payment.
```
Comment (Jason-Zhang9309) [inline docs/resources/cbh_instance.md]: this too

File: docs/resources/cbh_instance.md [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+  This parameter is mandatory if `charging_mode` is set to **prePaid**.
+
+  Changing this parameter will create a new resource.
+
+* `auto_renew` - (Required, String, ForceNew) Specifies whether auto renew is enabled.
+  Valid values are "true" and "false". Defaults to **false**.
+
+  Changing this parameter will create a new resource.
+
+* `subscription_num` - (Required, Int, ForceNew) Specifies the number of the subscription.
```
Comment (Jason-Zhang9309) [inline docs/resources/cbh_instance.md]: it seems the value of this parameter can only be `1`，how about just hide it

File: docs/resources/cbh_instance.md (hunk 2 above)
Comment (houpeng80) [reply docs/resources/cbh_instance.md]: done, it has been deleted

File: docs/resources/cbh_instance.md (hunk 3 above)
Comment (houpeng80) [reply docs/resources/cbh_instance.md]: done, it has been deleted

File: docs/resources/cbh_instance.md (hunk 4 above)
Comment (houpeng80) [reply docs/resources/cbh_instance.md]: Vdone, it has been deleted

File: docs/resources/cbh_instance.md (hunk 5 above)
Comment (houpeng80) [reply docs/resources/cbh_instance.md]: done

File: docs/resources/cbh_instance.md [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+variable "security_group_id" {}
+variable "availability_zone" {}
+variable "region" {}
+
+resource "huaweicloud_cbh_instance" "test" {
+  flavor_id         = var.flavor_id
+  name              = "cbh_instance_test"
+  vpc_id            = var.vpc_id
+  availability_zone = var.availability_zone
+  region            = var.region
```
Comment (ShiChangkuo) [inline docs/resources/cbh_instance.md]: usually, we does not use `region` in example usage, please delete it.

File: docs/resources/cbh_instance.md [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+* `ip_address` - (Optional, String) Specifies the IP address of the subnet.
+
+* `public_ip` - (Optional, String)
+  The [PublicIP](#CBHInstance_PublicIP) structure is documented below.
+
+* `number` - (Optional, Int, ForceNew) Specifies the elastic count.
+
+  Changing this parameter will create a new resource.
+
+* `root_volume` - (Optional, String, ForceNew)
```
Comment (ShiChangkuo) [inline docs/resources/cbh_instance.md]: the type of root_volume is wrong

File: docs/resources/cbh_instance.md [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+
+  Changing this parameter will create a new resource.
+  The [RootVolume](#CBHInstance_RootVolume) structure is documented below.
+
+* `data_volume` - (Optional, String, ForceNew)
```
Comment (ShiChangkuo) [inline docs/resources/cbh_instance.md]: the type of data_volume is wrong

File: docs/resources/cbh_instance.md [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+  The options are as follows: **create**, **renewals** and **change**.
+
+<a name="CBHInstance_PublicIP"></a>
+The `PublicIP` block supports:
+
+* `id` - (Optional, String) Specifies the ID of the elastic IP.
+
+* `address` - (Optional, String) Specifies the elastic IP address.
+
+* `eip` - (Optional, String)
```
Comment (ShiChangkuo) [inline docs/resources/cbh_instance.md]: the type of EIP is wrong

File: docs/resources/cbh_instance.md [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+  The [Eip](#CBHInstance_Eip) structure is documented below.
+
+<a name="CBHInstance_Eip"></a>
+The `PublicIPEip` block supports:
+
+* `type` - (Optional, String) Specifies the type of EIP.
+
+* `flavor_id` - (Optional, String) Specifies the product ID of the IP associated with.
+
+* `bandwidth` - (Optional, String)
```
Comment (ShiChangkuo) [inline docs/resources/cbh_instance.md]: the type of bandwidth is wrong

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go [hunk 11]
```diff
@@ -0,0 +1,1234 @@
...
+			"availability_zone": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the availability zone name.`,
+			},
+			"hx_password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
```
Comment (ShiChangkuo) [inline huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go:76]: Add `Sensitive: true`

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go [hunk 12]
```diff
@@ -0,0 +1,1234 @@
...
+				Description: `Specifies a image ID.`,
+			},
+			"user_data": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Specifies the inject user data.`,
+			},
+			"password": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go:131]: Add `Sensitive: true`

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go [hunk 13]
```diff
@@ -0,0 +1,1234 @@
...
+		Schema: map[string]*schema.Schema{
+			"size": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the size of the bandwidth.`,
+			},
+			"share_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the share type. Only PER is supported noe.`,
```
Comment (ShiChangkuo) [inline huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go]:
```suggestion
				Description: `Specifies the share type. Only PER is supported.`,
```

File: docs/resources/cbh_instance.md (hunk 6 above)
Comment (houpeng80) [reply docs/resources/cbh_instance.md]: done

File: docs/resources/cbh_instance.md (hunk 7 above)
Comment (houpeng80) [reply docs/resources/cbh_instance.md]: done

File: docs/resources/cbh_instance.md (hunk 8 above)
Comment (houpeng80) [reply docs/resources/cbh_instance.md]: done

File: docs/resources/cbh_instance.md (hunk 9 above)
Comment (houpeng80) [reply docs/resources/cbh_instance.md]: done

File: docs/resources/cbh_instance.md (hunk 10 above)
Comment (houpeng80) [reply docs/resources/cbh_instance.md]: done

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go (hunk 11 above)
Comment (houpeng80) [reply huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go:76]: done

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go (hunk 12 above)
Comment (houpeng80) [reply huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go:131]: done

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go (hunk 13 above)
Comment (houpeng80) [reply huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go]: done

**Pull request merged**